libp2p/io/abc.py

```diff
@@ -0,0 +1,34 @@
+from abc import ABC, abstractmethod
+
+
+class Closer(ABC):
+    async def close(self) -> None:
+        ...
+
+
+class Reader(ABC):
+    @abstractmethod
+    async def read(self, n: int = -1) -> bytes:
+        ...
+
+
+class Writer(ABC):
+    @abstractmethod
+    async def write(self, data: bytes) -> int:
```

Comment:
Though in Go implementation `write` returns `int` which indicates how many bytes are written, IMO we can just return `None` in Python because so far I haven't seen any use case of this return value. And also, `asyncio.StreamWriter.write` returns `None`. What do you think about it?

Reply:
i don't know enough about how libp2p uses the data to say.... in general a low-level write call may only write some of the data and the return information tells you that... it could be that we are working w/ abstractions that just guarantee an entire write but in general may not in the future (i.e. different async library or more performant code).... if it becomes obvious we don't need this then i'm very fine removing it :)